circuits/src/lib.rs

```diff
@@ -578,6 +587,30 @@ impl CommitPublic for Scalar {
     }
 }
 
+impl CommitPublic for bool {
+    type VarType = Variable;
+    type ErrorType = ();
+
+    fn commit_public<CS: RandomizableConstraintSystem>(
+        &self,
+        cs: &mut CS,
+    ) -> Result<Self::VarType, Self::ErrorType> {
+        Ok(cs.commit_public(Scalar::from(*self as u8)))
+    }
+}
+
+impl CommitPublic for () {
+    type VarType = ();
+    type ErrorType = ();
+
+    fn commit_public<CS: RandomizableConstraintSystem>(
+        &self,
+        _cs: &mut CS,
+    ) -> Result<Self::VarType, Self::ErrorType> {
+        Ok(())
+    }
+}
+
```

Comment:
ngl this feels pretty gross, but it felt overkill to make statement types to wrap `bool` and `()` in the places where these are used as statements